### PR TITLE
Fix replication regression

### LINKED ISSUE #11340: File replication stopped work for bucket where enabled DeleteReplication and DeleteMarkerReplication
I have buckets in the two datacenter with enabled replication between each other.
Replication of files stopped working after enabling DeleteReplication and DeleteMarkerReplication feature.
```
mc replicate edit local/test --id "bt8u4bnckkkgn5mck280" --replicate "delete,delete-marker"
```

I've tried to remove replication and create it again. This action did not help.

## Current Behavior
Currently I see this file replication statues:
```
# mc stat local/test/file-2.yaml
Name      : zbx_export_templates-2.yaml
Date      : 2021-01-23 09:20:01 CET
Size      : 15 KiB
ETag      : 6c8ced0e9f644b94542defeff66372c3
VersionID : 9044152a-7c52-4142-be07-fe9600f4b274
Type      : file
Metadata  :
  Content-Type: application/x-yaml
Replication Status: FAILED


# mc stat local/test/file-36.xml
Name      : zbx_export_templates-36.xml
Date      : 2021-01-24 16:50:01 CET
Size      : 34 KiB
ETag      : 333ca4bb0e6a53bb2d3337765a0e03cb
VersionID : 4183c14d-5448-4aaa-b0c0-cb1878b25ca1
Type      : file
Metadata  :
  Content-Type: text/xml
Replication Status: PENDING
```

Replication parameters:
```
# mc replicate export local/test
{
 "Rules": [
  {
   "ID": "c06pctnckkkjebv3ihog",
   "Status": "Enabled",
   "Priority": 1,
   "DeleteMarkerReplication": {
    "Status": "Enabled"
   },
   "DeleteReplication": {
    "Status": "Enabled"
   },
   "Destination": {
    "Bucket": "arn:aws:s3:::test",
    "StorageClass": "STANDARD"
   },
   "Filter": {
    "And": {},
    "Tag": {}
   },
   "SourceSelectionCriteria": {
    "ReplicaModifications": {
     "Status": ""
    }
   }
  }
 ],
 "Role": "arn:minio:replication:ams01:ec22934bb052ca100db62c883efba31df89547bd2705e521dded84fbccac40bb:test"
}
```

## Regression
Yes

## My Environment
minio: RELEASE.2021-01-16T02-19-44Z inside docker container

OS: CentOS Linux release 8.2.2004
Kernel: 4.18.0-193.14.2.el8_2.x86_64

### PR DESCRIPTION
**Pull request by poornas**
From PR #11165 due to incorrect
proxying for 2 way replication even
when the object was not yet replicated

Also fixing metadata comparisons when
deciding to do full replciation vs metadata copy.

Also Fixes #11340

## How to test this PR?
Turn on 2 way replication and do some uploads to source. `mc admin trace` should show a PUT operation on the dest, not a COPY in master due to incorrect proxying.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation updated
- [ ] Unit tests added/updated

### CODE REVIEW

File: cmd/bucket-replication.go [hunk 1]
```diff
@@ -129,19 +129,27 @@ func mustReplicater(ctx context.Context, bucket, object string, meta map[string]
 		opts.UserTags = tagStr
 	}
 	tgt := globalBucketTargetSys.GetRemoteTargetClient(ctx, cfg.RoleArn)
-	if tgt == nil || tgt.isOffline() {
```
Comment (harshavardhana) [inline cmd/bucket-replication.go:132]: why not invert this logic? @poornas 

```
if tgt != nil && !tgt.isOffline()  {
    return cfg.Replicate(opts), tgt.replicateSync
}
return cfg.Replicate(opts), false
```

File: cmd/bucket-replication.go [hunk 2]
```diff
@@ -730,7 +760,9 @@ func proxyHeadToRepTarget(ctx context.Context, bucket, object string, opts Objec
 	if tgt == nil || tgt.isOffline() {
 		return nil, oi, false, fmt.Errorf("missing target")
```
Comment (harshavardhana) [inline cmd/bucket-replication.go]:
```suggestion
		return nil, oi, false, fmt.Errorf("target is offline or not configured")
```

File: cmd/object-api-options.go [hunk 3]
```diff
@@ -66,7 +66,7 @@ func getDefaultOpts(header http.Header, copySource bool, metadata map[string]str
 	if crypto.S3.IsRequested(header) || (metadata != nil && crypto.S3.IsEncrypted(metadata)) {
 		opts.ServerSideEncryption = encrypt.NewSSE()
 	}
-	opts.ProxyRequest = header.Get(xhttp.MinIOSourceProxyRequest) == "true"
+	opts.ProxyRequest = header.Get(xhttp.MinIOSourceProxyRequest)
```
Comment (harshavardhana) [inline cmd/object-api-options.go]:
```suggestion
	opts.ProxyRequest = header.Get(xhttp.MinIOSourceProxyRequest)
```

why change this? @poornas

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: cmd/bucket-replication.go (hunk 1 above)
Comment (poornas) [reply cmd/bucket-replication.go:132]: done

File: cmd/object-api-options.go (hunk 3 above)
Comment (poornas) [reply cmd/object-api-options.go]: reverted back to using bool

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#890ba8b7-fa18-4596-a250-dd22b1a891b8) |
<a name="890ba8b7-fa18-4596-a250-dd22b1a891b8"></a>
### 11356-0a263c8/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c3.minio.io:30517
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 6b005aa18180:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 10 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 3 minutes and 21 seconds
(4/15) Running aws-sdk-ruby tests ... done in 25 seconds
(5/15) Running awscli tests ... done in 2 minutes and 57 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 33 minutes and 49 seconds
(8/15) Running minio-dotnet tests ... done in 4 minutes and 27 seconds
(9/15) Running minio-go tests ... done in 23 minutes and 41 seconds
(10/15) Running minio-java tests ... FAILED in 9 minutes and 23 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[user metadata]",
  "duration": 175,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = AuthenticationFailed, message = -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=AuthenticationFailed) =====\nDescription=Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\nRequestId:8113f196-e01e-003d-4e52-f458c8000000\nTime:2021-01-27T02:18:59.9516490Z, Details: \n   AuthenticationErrorDetail: The MAC signature found in the HTTP request 'P4sVSLcPvIJFh+bxSNI6kMChPBQmDaqHnJ5lE8nQA8w=' is not the same as any computed signature. Server used following string to sign: 'PUT\n\n\n128\n\napplication/xml\n\n\n\n\n\n\nx-ms-blob-cache-control:\nx-ms-blob-content-disposition:\nx-ms-blob-content-encoding:\nx-ms-blob-content-language:\nx-ms-blob-content-type:application/octet-stream\nx-ms-client-request-id:0420ceb5-7e9c-40a0-50df-204be71039db\nx-ms-date:Wed, 27 Jan 2021 02:18:59 GMT\nx-ms-meta-my_header1:a   b   c\nx-ms-meta-my_header2:\"a   b   c\"\nx-ms-meta-my_project:Project One\nx-ms-meta-my_unicode_tag:å•†å“\nx-ms-version:2019-02-02\n/minioazure/minio-java-test-1q8h8ul/minio-java-test-2l70q3o\ncomp:blocklist\ntimeout:1501'.\n   Code: AuthenticationFailed\n   PUT https://minioazure.blob.core.windows.net/minio-java-test-1q8h8ul/minio-java-test-2l70q3o?comp=blocklist&timeout=1501\n   Authorization: REDACTED\n   Content-Length: [128]\n   Content-Type: [application/xml]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2021-01-27T00:54:26Z]\n   X-Ms-Blob-Cache-Control: []\n   X-Ms-Blob-Content-Disposition: []\n   X-Ms-Blob-Content-Encoding: []\n   X-Ms-Blob-Content-Language: []\n   X-Ms-Blob-Content-Type: [application/octet-stream]\n   X-Ms-Client-Request-Id: [0420ceb5-7e9c-40a0-50df-204be71039db]\n   X-Ms-Date: [Wed, 27 Jan 2021 02:18:59 GMT]\n   X-Ms-Meta-My_header1: [a   b   c]\n   X-Ms-Meta-My_header2: [\"a   b   c\"]\n   X-Ms-Meta-My_project: [Project One]\n   X-Ms-Meta-My_unicode_tag: [商品]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 403 Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\n   Content-Length: [1092]\n   Content-Type: [application/xml]\n   Date: [Wed, 27 Jan 2021 02:18:59 GMT]\n   Server: [Microsoft-HTTPAPI/2.0]\n   X-Ms-Error-Code: [AuthenticationFailed]\n   X-Ms-Request-Id: [8113f196-e01e-003d-4e52-f458c8000000]\n\n\n, bucketName = minio-java-test-1q8h8ul, objectName = minio-java-test-2l70q3o, resource = /minio-java-test-1q8h8ul/minio-java-test-2l70q3o, requestId = 165DF549FFE326B1, hostId = 6afd77c2-4b53-45a5-b8ed-db6df08210a7)\nrequest={method=PUT, url=http://minio-c3.minio.io:30517/minio-java-test-1q8h8ul/minio-java-test-2l70q3o, headers=x-amz-meta-My-Unicode-Tag: 商品\nx-amz-meta-My-Project: Project One\nx-amz-meta-My-header1: a   b   c\nx-amz-meta-My-Header2: \"a   b   c\"\nContent-Type: application/octet-stream\nHost: minio-c3.minio.io:30517\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20210127T021859Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210127/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-meta-my-header1;x-amz-meta-my-header2;x-amz-meta-my-project;x-amz-meta-my-unicode-tag, Signature=*REDACTED*\n}\nresponse={code=403, headers=Accept-Ranges: bytes\nContent-Length: 3086\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 165DF549FFE326B1\nX-Xss-Protection: 1; mode=block\nDate: Wed, 27 Jan 2021 02:18:59 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.putObject(MinioClient.java:4547), io.minio.MinioClient.putObject(MinioClient.java:2713), io.minio.MinioClient.putObject(MinioClient.java:2830), FunctionalTest.testPutObject(FunctionalTest.java:763), FunctionalTest.putObject(FunctionalTest.java:890), FunctionalTest.runObjectTests(FunctionalTest.java:3751), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(10/15) Running minio-js tests ... done in 4 minutes and 22 seconds
(11/15) Running minio-py tests ... FAILED in 23 minutes and 24 seconds
{
  "name": "minio-py:test_select_object_content",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-a94da7c2-8d20-4f05-a423-a33728d1f063",
    "object_name": "test.csv"
  },
  "message": "InternalError: file not found",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 285, in test_select_object_content\n    for data_bytes in data.stream(10*KB):\n  File \"/usr/local/lib/python3.6/dist-packages/minio/select.py\", line 444, in stream\n    if self._read() <= 0:\n  File \"/usr/local/lib/python3.6/dist-packages/minio/select.py\", line 406, in _read\n    headers.get(\":error-code\"), headers.get(\":error-message\"),\nminio.error.MinioException: InternalError: file not found\n",
  "duration": 1308
}
(11/15) Running s3cmd tests ... done in 10 minutes and 24 seconds
(12/15) Running s3select tests ... done in 1 minutes and 2 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/bucket-replication.go [hunk 4]
```diff
@@ -701,10 +738,11 @@ func isProxyable(ctx context.Context, bucket string) bool {
 	dest := cfg.GetDestination()
 	return dest.Bucket == bucket
 }
+
 func proxyHeadToRepTarget(ctx context.Context, bucket, object string, opts ObjectOptions) (tgt *TargetClient, oi ObjectInfo, proxy bool, err error) {
 	// this option is set when active-active replication is in place between site A -> B,
 	// and site B does not have the object yet.
-	if opts.ProxyRequest { // true only when site B sets MinIOSourceProxyRequest header
+	if opts.ProxyRequest || (opts.ProxyHeaderSet && !opts.ProxyRequest) { // true only when site B sets MinIOSourceProxyRequest header
```
Comment (vadmeste) [inline cmd/bucket-replication.go:745]: Is storage class concept included in replication as well? That's why we are proxying head requests ?

File: cmd/bucket-replication.go [hunk 5]
```diff
@@ -728,14 +766,16 @@ func proxyHeadToRepTarget(ctx context.Context, bucket, object string, opts Objec
 	}
 	tgt = globalBucketTargetSys.GetRemoteTargetClient(ctx, cfg.RoleArn)
 	if tgt == nil || tgt.isOffline() {
-		return nil, oi, false, fmt.Errorf("missing target")
+		return nil, oi, false, fmt.Errorf("target is offline or not configured")
+	}
+	if !tgt.replicateSync {
```
Comment (vadmeste) [inline cmd/bucket-replication.go]: What's the relation between tgt.replicateSync and proxyHeadToRepTarget() ? Why we are quitting early here ?

**Review by vadmeste — COMMENTED** [review]
Some questions, but tested and it works

File: cmd/bucket-replication.go (hunk 4 above)
Comment (poornas) [reply cmd/bucket-replication.go:745]: purpose of proxying when 2 way replication is set up is for automatic failover if one cluster has the object but not yet replicated to other. Not sure what role storage class plays here?

File: cmd/bucket-replication.go (hunk 5 above)
Comment (poornas) [reply cmd/bucket-replication.go]: Actually, this check is not required - removing it.

File: cmd/bucket-replication.go (hunk 4 above)
Comment (vadmeste) [reply cmd/bucket-replication.go:745]: > purpose of proxying when 2 way replication is set up is for automatic failover if one cluster has the object but not yet replicated to other. Not sure what role storage class plays here?

okay, I understood now the relation between replication & proxying

**Review by vadmeste — APPROVED** [review]
LGTM & tested

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#72c747e9-5181-4727-8dc5-c2218599de84) |
<a name="72c747e9-5181-4727-8dc5-c2218599de84"></a>
### 11356-941f750/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-c2.minio.io:31440
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp dbab526f51b4:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 11 seconds
(2/15) Running aws-sdk-java tests ... done in 1 seconds
(3/15) Running aws-sdk-php tests ... done in 3 minutes and 49 seconds
(4/15) Running aws-sdk-ruby tests ... done in 21 seconds
(5/15) Running awscli tests ... done in 3 minutes and 3 seconds
(6/15) Running healthcheck tests ... done in 0 seconds
(7/15) Running mc tests ... done in 39 minutes and 37 seconds
(8/15) Running minio-dotnet tests ... done in 4 minutes and 12 seconds
(9/15) Running minio-go tests ... done in 26 minutes and 54 seconds
(10/15) Running minio-java tests ... FAILED in 9 minutes and 18 seconds
{
  "name": "minio-java",
  "function": "putObject()",
  "args": "[user metadata]",
  "duration": 169,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = AuthenticationFailed, message = -> github.com/Azure/azure-storage-blob-go/azblob.newStorageError, github.com/Azure/azure-storage-blob-go@v0.10.0/azblob/zc_storage_error.go:42\n===== RESPONSE ERROR (ServiceCode=AuthenticationFailed) =====\nDescription=Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\nRequestId:28a157f7-e01e-0050-65e3-f4f2e6000000\nTime:2021-01-27T19:35:15.5158279Z, Details: \n   AuthenticationErrorDetail: The MAC signature found in the HTTP request 'Oy5KNJJ3h8jCwOLOofTNqfU5be1e/Lg/u6oOCJkEtHg=' is not the same as any computed signature. Server used following string to sign: 'PUT\n\n\n128\n\napplication/xml\n\n\n\n\n\n\nx-ms-blob-cache-control:\nx-ms-blob-content-disposition:\nx-ms-blob-content-encoding:\nx-ms-blob-content-language:\nx-ms-blob-content-type:application/octet-stream\nx-ms-client-request-id:8beaa75b-2a04-467a-45d5-bb92a10c1f1d\nx-ms-date:Wed, 27 Jan 2021 19:35:15 GMT\nx-ms-meta-my_header1:a   b   c\nx-ms-meta-my_header2:\"a   b   c\"\nx-ms-meta-my_project:Project One\nx-ms-meta-my_unicode_tag:å•†å“\nx-ms-version:2019-02-02\n/minioazure/minio-java-test-1mil7kh/minio-java-test-1hilfrb\ncomp:blocklist\ntimeout:1501'.\n   Code: AuthenticationFailed\n   PUT https://minioazure.blob.core.windows.net/minio-java-test-1mil7kh/minio-java-test-1hilfrb?comp=blocklist&timeout=1501\n   Authorization: REDACTED\n   Content-Length: [128]\n   Content-Type: [application/xml]\n   User-Agent: [APN/1.0 MinIO/1.0 MinIO/2021-01-27T18:01:26Z]\n   X-Ms-Blob-Cache-Control: []\n   X-Ms-Blob-Content-Disposition: []\n   X-Ms-Blob-Content-Encoding: []\n   X-Ms-Blob-Content-Language: []\n   X-Ms-Blob-Content-Type: [application/octet-stream]\n   X-Ms-Client-Request-Id: [8beaa75b-2a04-467a-45d5-bb92a10c1f1d]\n   X-Ms-Date: [Wed, 27 Jan 2021 19:35:15 GMT]\n   X-Ms-Meta-My_header1: [a   b   c]\n   X-Ms-Meta-My_header2: [\"a   b   c\"]\n   X-Ms-Meta-My_project: [Project One]\n   X-Ms-Meta-My_unicode_tag: [商品]\n   X-Ms-Version: [2019-02-02]\n   --------------------------------------------------------------------------------\n   RESPONSE Status: 403 Server failed to authenticate the request. Make sure the value of Authorization header is formed correctly including the signature.\n   Content-Length: [1092]\n   Content-Type: [application/xml]\n   Date: [Wed, 27 Jan 2021 19:35:15 GMT]\n   Server: [Microsoft-HTTPAPI/2.0]\n   X-Ms-Error-Code: [AuthenticationFailed]\n   X-Ms-Request-Id: [28a157f7-e01e-0050-65e3-f4f2e6000000]\n\n\n, bucketName = minio-java-test-1mil7kh, objectName = minio-java-test-1hilfrb, resource = /minio-java-test-1mil7kh/minio-java-test-1hilfrb, requestId = 165E2DD660417345, hostId = c4d4245d-28bd-4c36-a10d-58974a9be285)\nrequest={method=PUT, url=http://minio-c2.minio.io:31440/minio-java-test-1mil7kh/minio-java-test-1hilfrb, headers=x-amz-meta-My-Unicode-Tag: 商品\nx-amz-meta-My-Project: Project One\nx-amz-meta-My-header1: a   b   c\nx-amz-meta-My-Header2: \"a   b   c\"\nContent-Type: application/octet-stream\nHost: minio-c2.minio.io:31440\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.0.3\nContent-MD5: A9oFTxee7YVcJ9fWsgQeKg==\nx-amz-content-sha256: 1ff7959f86334ddc5c188a5083268f600146328b2b6c5185e75bf7d9387d6b74\nx-amz-date: 20210127T193515Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20210127/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date;x-amz-meta-my-header1;x-amz-meta-my-header2;x-amz-meta-my-project;x-amz-meta-my-unicode-tag, Signature=*REDACTED*\n}\nresponse={code=403, headers=Accept-Ranges: bytes\nContent-Length: 3086\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nVary: Origin\nX-Amz-Request-Id: 165E2DD660417345\nX-Xss-Protection: 1; mode=block\nDate: Wed, 27 Jan 2021 19:35:15 GMT\n}\n >>> [io.minio.MinioClient.execute(MinioClient.java:775), io.minio.MinioClient.putObject(MinioClient.java:4547), io.minio.MinioClient.putObject(MinioClient.java:2713), io.minio.MinioClient.putObject(MinioClient.java:2830), FunctionalTest.testPutObject(FunctionalTest.java:763), FunctionalTest.putObject(FunctionalTest.java:890), FunctionalTest.runObjectTests(FunctionalTest.java:3751), FunctionalTest.runTests(FunctionalTest.java:3783), FunctionalTest.main(FunctionalTest.java:3927)]"
}
(10/15) Running minio-js tests ... done in 3 minutes and 56 seconds
(11/15) Running minio-py tests ... done in 22 minutes and 43 seconds
(12/15) Running s3cmd tests ... done in 9 minutes and 53 seconds
(13/15) Running s3select tests ... done in 1 minutes and 7 seconds
(14/15) Running security tests ... done in 0 seconds

Executed 14 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally

**Pull request merged**